### ISSUE TITLE
[HOLD RNW #2542] [$2000] mWeb - Login - Cursor lags when entering magic code

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#19220]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Start mobile Safari and open staging.new.expensify.com
2. Enter any email in login field
3. Start entering the code from the email you received

## Expected Result:
The cursor moves to the next digit at the same time as it is entered

## Actual Result:
The cursor is late and stuck on the first digit

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [x] iOS / Safari
- [ ] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.16.3

**Reproducible in staging?:** yes

**Reproducible in production?:** yes

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://platform.applause.com/services/links/v1/external/103e5a538ec9bd253e83dbddbfb62ec9cdb3959a177c9fc99441959598fa9ed4

**Expensify/Expensify Issue URL:**

**Issue reported by:** Applause - Internal Team

**Slack conversation:**

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01ef374cfdc2221640</li>
        <li>Upwork Job ID: 1661480889752801280</li>
        <li>Last Price Increase: 2023-06-16</li>
    </ul>
</details>

**Comment by MitchExpensify** [comment]
I was undable to reproduce this on my flight today, will try again tomorrow

**Comment by MitchExpensify** [comment]
I was undable to reproduce this on my flight today, will try again tomorrow

**Comment by MitchExpensify** [comment]
I was undable to reproduce this on my flight today, will try again tomorrow

**Comment by MitchExpensify** [comment]
I was undable to reproduce this on my flight today, will try again tomorrow

**Comment by MitchExpensify** [comment]
I was undable to reproduce this on my flight today, will try again tomorrow

**Comment by MitchExpensify** [comment]
Reproduced using the steps in the issue, labelling external

**Comment by s77rt** [comment]
Taking this as C+

**Comment by s77rt** [comment]
Is this still reproducible? I don't even see the cursor :sweat_smile:.

![Screenshot from 2023-05-27 01-08-03](https://github.com/Expensify/App/assets/16493223/0b5d1f02-b573-4cac-921f-2b3c705a6935)

**Comment by s77rt** [comment]
@getusha Is this on `main` and what iOS version is this? (if it's 16.4, can you try with others?)

**Comment by s77rt** [comment]
@getusha Is the issue also reproducible in other iOS versions? Version 16.4 seems buggy at the moment

**Comment by s77rt** [comment]
@getusha We have this issue https://github.com/Expensify/App/issues/18218 that relates to the magic code input and it is reproducible only on iOS 16.4. Just worth checking if this is a similar one.
Side note: The uploaded videos in that issue does not seem to have a visible cursor either

**Comment by s77rt** [comment]
@getusha Yeah something weird is going on :sweat_smile:. I tested on iOS 16.2, can you test on that or on 16.3.

**Comment by sakluger** [comment]
@s77rt assigned you as the C+, thanks for helping with this one!

**Comment by s77rt** [comment]
I think we still need to understand the root cause to determine if this is a dupe. (The other issue won't be closed. It's opened specifically for iOS 16.4)

**Comment by s77rt** [comment]
They seem to have the same reproduction circumstances but not necessary the same root cause. We don't know the root cause of any so it's hard to tell if this is a dupe.
Also the matter of having a cursor or not is concerning, we don't know why the cursor is visible in some cases and others not.

**Comment by MitchExpensify** [comment]
Whats the latest on this issue from your perspective @s77rt ?

**Comment by s77rt** [comment]
@MitchExpensify I was not able to reproduce the bug on iOS 16.2 (the cursor is not even visible). I think we need more investigation on this one. Also there is a possibility that the root cause is linked to https://github.com/Expensify/App/issues/18218.

**Comment by MitchExpensify** [comment]
Got it, thank you!

**Comment by s77rt** [comment]
Melvin, we are still looking for deeper explanation on the cursor behaviour and if it's related to https://github.com/Expensify/App/issues/18218

**Comment by s77rt** [comment]
Not overdue. Still looking for proposals

**Comment by MitchExpensify** [comment]
Friendly bump on the above @getusha

**Comment by s77rt** [comment]
Still lookng for proposals

**Comment by puneetlath** [comment]
@MitchExpensify I'll be OOO this week, so if there's a proposal that ends up needing approval, please use the Engineering label to get someone else assigned. Thanks!

**Comment by MitchExpensify** [comment]
Waiting on proposals

**Comment by MitchExpensify** [comment]
Going to manually double this to see if it brings some more proposals.. Not sure why its not auto doubling

**Comment by MitchExpensify** [comment]
Posted to Slack about the double up here https://expensify.slack.com/archives/C01GTK53T8Q/p1686878545866609

**Comment by aswin-s** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

On iOS Mobile, magic code input shows cursor in the first input field and lags behind the active input.


### What is the root cause of that problem?

On iOS Safari cursor is not hidden where as in other platforms cursor is hidden by outline:none style.


### What changes do you think we should make in order to solve the problem?

iOS mobile needs additional css style `caret-style: none` to be added to hide the cursor in input.

Add style to below line. https://github.com/Expensify/App/blob/main/src/styles/styles.js#L2484

```js
magicCodeInputTransparent: {
  color: 'transparent',
  // Add style
  caretColor: 'transparent',
},
```
### What alternative solutions did you explore? (Optional)

**Reminder:** Please use plain English, be brief and avoid jargon. Feel free to use images, charts or pseudo-code if necessary. Do not post large multi-line diffs or write walls of text. Do not create PRs unless you have been hired for this job.

None

**Comment by s77rt** [comment]
@honnamkuan Thanks for the proposal. Your RCA is correct. The suggested fix makes sense but let's try to avoid adding platform dependent style for now (`caretColor` is not supported by native app). Can we do the following:
1. Revert https://github.com/Expensify/App/pull/18771
2. Fix https://github.com/Expensify/App/issues/18482 in another way?


___

_@aswin-s I'm reviewing @honnamkuan proposal first based on the timeline on the other linked issue._

**Comment by s77rt** [comment]
@aswin-s Thanks for the proposal. Your RCA is correct. The alternative solution you proposed of fixing this upstream in RNW (adding support for `caretHidden`) looks good. to me. But before moving forward with making another fix it would better to evaluate the process of taking a step back and fixing an older issue (as mentioned in the comment above: revert PR and fix the old issue).

**Comment by s77rt** [comment]
Not overdue. We have valid proposals but looking for another approach (the take a step back approach).

**Comment by s77rt** [comment]
@aswin-s Do you have a plan for adding support for `caretHidden` in RNW?

**Comment by aswin-s** [comment]
@s77rt I've raised [an issue](https://github.com/necolas/react-native-web/issues/2541) at react-native-web citing lacking support for `caretHidden` prop. I've also [raised a PR](https://github.com/necolas/react-native-web/pull/2542) which will add support for the prop. The fix could be tested from below URL.

https://codesandbox.io/p/sandbox/react-native-web-examples-c8gl3p

Since we are using a fork of react-native-web, the commit could be cherry picked and merged to [@expensify/react-native-web](https://github.com/Expensify/react-native-web) without waiting for PR to get approved and merged upstream.

**Comment by s77rt** [comment]
@aswin-s I don't see the urgency here. Let's work this one on RNW repo first. I have added a comment there. Lets also wait for maintainers review.

**Comment by aswin-s** [comment]
@s77rt PR got merged upstream. We need to cherry pick the commit or sync the Expensify fork for RNW.

**Comment by aswin-s** [comment]
Able to reproduce on staging and production urls. Also the suggestion above was to avoid platform specific styling.

**Comment by s77rt** [comment]
@puneetlath Can you please help in updating RNW here?

**Comment by puneetlath** [comment]
Yep, I can open an issue for it. What version do we want to update to? 0.19?

**Comment by s77rt** [comment]
@puneetlath To at least 0.19.6 or to just cherry pick https://github.com/necolas/react-native-web/commit/1c376dbb0c8f223370814c4b21fbf57d25ef79f2

**Comment by s77rt** [comment]
Can you also remove the `Internal` label here

**Comment by MitchExpensify** [comment]
Is this ready for the Help wanted label? @s77rt

**Comment by s77rt** [comment]
@MitchExpensify No, we don't need that label. We have a solution already we need to update RNW to proceed

**Comment by MitchExpensify** [comment]
Thanks @s77rt !

**Comment by puneetlath** [comment]
@MitchExpensify could you create a separate issue for upgrading to RNW v0.19.6? And maybe we should offer it to an expert firm like Callstack to take on?

**Comment by puneetlath** [comment]
Ah, thanks for the link @aimane-chnaif.

**Comment by puneetlath** [comment]
@MitchExpensify in that case, I think we should go ahead and compensate @aswin-s and @s77rt for their work here, but leave this issue open and on hold for https://github.com/Expensify/App/issues/16660 to track that we do get the upstream fix incorporated.

**Comment by MitchExpensify** [comment]
@puneetlath can do! Do you agree $2000 is fair for both contributors given the work here so far?

**Comment by puneetlath** [comment]
Given that's the bounty on the issue, seems fair enough to me!

**Comment by MitchExpensify** [comment]
Thanks! Invite sent to @aswin-s and @s77rt for payment https://www.upwork.com/jobs/~01ef374cfdc2221640

**Comment by aswin-s** [comment]
Offer accepted

**Comment by s77rt** [comment]
@MitchExpensify Accepted! Thanks!

**Comment by MitchExpensify** [comment]
Ugh, sorry team! I don't know why Upwork won't let me issue payment on that job I posted earlier so I've created another and invited you to it here https://www.upwork.com/jobs/~01eab965d5d9f6a41a Thanks for your patience @s77rt & @aswin-s

**Comment by s77rt** [comment]
@MitchExpensify No problem! Accepted the new offer

**Comment by MitchExpensify** [comment]
Paid! Thanks

**Comment by s77rt** [comment]
Still on hold for https://github.com/Expensify/App/issues/16660

**Comment by puneetlath** [comment]
Still on hold.

**Comment by s77rt** [comment]
Not overdue. This is still on hold

**Comment by s77rt** [comment]
Still on hold

**Comment by s77rt** [comment]
https://github.com/Expensify/App/issues/16660 is completed and the bug is fixed. We can close this. cc @MitchExpensify
